Sources/CryptoSwift/AES.swift

```diff
@@ -120,11 +133,21 @@ public final class AES: BlockCipher {
     /// - throws: AES.Error
     ///
     /// - returns: Instance
-    public init(key: Array<UInt8>, blockMode: BlockMode, padding: Padding = .pkcs7) throws {
+    public init(key: Array<UInt8>, blockMode: BlockMode, padding: Padding = .pkcs7, variant: Variant? = nil) throws {
         self.key = Key(bytes: key)
         self.blockMode = blockMode
         self.padding = padding
-        keySize = self.key.count
+        self.keySize = self.key.count
+
+        if let variant = variant {
+            guard variant.keySize <= keySize else {
```

Comment:
!=

it's not enough to check if it's shorter, is it?

Reply:
If the size has to be the same, my PR is nonsense : It is intended to use a 64 length key with 128 variant (which has to be a 16 length key based on the current version)

Reply:
you can't just use 64 bytes when you need 128 bytes. What's the point of that?

Reply:
the 128 variant use a 16 length key. I have a 64 length key, and since it was working on previous versions, I need to keep this key, for compatibility purpose

Reply:
We used the 0.7.2 version

Reply:
hm... When I'm trying to use 64 bytes length key with AES on 0.7.2 it crashes - that is kinda expected, although the error would be better.